mithril-common/benches/merkle_map.rs

```diff
@@ -0,0 +1,241 @@
+        .insert(
+            mk_map_key_to_prove.to_owned(),
+            mk_map_node_to_prove.to_owned(),
+        )
+        .unwrap();
+    mk_map_compressed.compute_proof(&leaves_to_prove).unwrap()
+}
+
+fn generate_merkle_map_proof(
+    block_ranges_nodes_iterator: impl Iterator<Item = (BlockRange, MKMapNode<BlockRange>)>,
+    mk_map_compressed: &MKMap<BlockRange, MKMapNode<BlockRange>>,
+    total_proofs: u64,
+) -> MKMapProof<BlockRange> {
+    let mut leaves_to_prove_all: Vec<MKTreeNode> = vec![];
+    let mut mk_map_compressed = mk_map_compressed.clone();
+    for (mk_map_key_to_prove, mk_map_node_to_prove) in &block_ranges_nodes_iterator
+        .take(total_proofs as usize)
+        .collect::<Vec<_>>()
+    {
+        let mktree_to_prove = if let MKMapNode::Tree(mktree_to_prove) = mk_map_node_to_prove {
+            mktree_to_prove
+        } else {
+            panic!("Expected MKMapNode::Tree");
+        };
+        let leaves_to_prove = mktree_to_prove
+            .leaves()
+            .into_iter()
+            .take(1)
+            .collect::<Vec<_>>();
+        leaves_to_prove_all.extend(leaves_to_prove);
+        mk_map_compressed
+            .insert(
+                mk_map_key_to_prove.to_owned(),
+                mk_map_node_to_prove.to_owned(),
+            )
+            .unwrap();
+    }
+    mk_map_compressed
+        .compute_proof(&leaves_to_prove_all)
+        .unwrap()
+}
+
+fn create_merkle_map_root_benches(c: &mut Criterion) {
+    let mut group = c.benchmark_group(format!(
+        "create_merkle_map_root(blocks_ranges_length={BLOCK_RANGE_LENGTH_BENCH},txs_per_block={TOTAL_TRANSACTIONS_PER_BLOCK})"
+    ));
+    for total_leaves in TOTAL_TRANSACTIONS_BENCHES.iter() {
+        let mk_trees_by_block_range_iterator = generate_block_ranges_nodes_iterator(
+            *total_leaves,
+            TOTAL_TRANSACTIONS_PER_BLOCK,
+            BLOCK_RANGE_LENGTH_BENCH,
+            1,
+        );
+        let mk_map_compressed = generate_merkle_map_compressed(mk_trees_by_block_range_iterator);
+        group.bench_with_input(
+            BenchmarkId::from_parameter(total_leaves),
+            total_leaves,
+            |b, &_total_leaves| {
+                b.iter(|| {
+                    let mk_trees_by_block_range_iterator = generate_block_ranges_nodes_iterator(
+                        *total_leaves,
+                        TOTAL_TRANSACTIONS_PER_BLOCK,
+                        BLOCK_RANGE_LENGTH_BENCH,
+                        1,
+                    );
+                    generate_merkle_map_root(mk_trees_by_block_range_iterator, &mk_map_compressed);
+                });
+            },
+        );
+    }
+    group.finish();
+}
+
+fn create_merkle_map_proof_benches(c: &mut Criterion) {
+    let mut group = c.benchmark_group(format!(
+        "create_merkle_map_proof_(blocks_ranges_length={BLOCK_RANGE_LENGTH_BENCH},txs_per_block={TOTAL_TRANSACTIONS_PER_BLOCK},txs_per_proof={MAX_TRANSACTIONS_PER_PROOF_BENCH})"
```

Comment:
This is same comment as before. The code is the same between `create_merkle_map_root_benches` and `create_merkle_map_proof_benches` but the max transaction per proof bench.
Would it make sense to extract a common function ?

Reply:
The code to benchmark is actually different when computing the root and when computing the proof so we'd better keep it separated.